src/monodroid/jni/monodroid-glue-internal.hh

```diff
@@ -10,10 +11,65 @@
 #include <mono/utils/mono-counters.h>
 #include <mono/metadata/profiler.h>
 
+#if defined (NET6)
+// NDEBUG causes robin_map.h not to include <iostream> which, in turn, prevents indirect inclusion of <mutex>. <mutex>
+// conflicts with our std::mutex definition in cppcompat.hh
+#if !defined (NDEBUG)
+#define NDEBUG
+#define NDEBUG_UNDEFINE
+#endif
+
+// hush some compiler warnings
+#if defined (__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunused-parameter"
+#endif // __clang__
+
+#include <tsl/robin_map.h>
+
+#if defined (__clang__)
+#pragma clang diagnostic pop
+#endif // __clang__
+
+#if defined (NDEBUG_UNDEFINE)
+#undef NDEBUG
+#undef NDEBUG_UNDEFINE
+#endif
+
+#include <mono/jit/mono-private-unstable.h>
+#include <mono/metadata/mono-private-unstable.h>
+
+// This should be defined in the public Mono headers
+typedef void * (*PInvokeOverrideFn) (const char *libraryName, const char *entrypointName);
+#endif
+
 namespace xamarin::android::internal
 {
 	class MonodroidRuntime
 	{
+#if defined (NET6)
+		using pinvoke_api_map = tsl::robin_map<
+			std::string,
+			void*,
+			std::hash<std::string>,
+			std::equal_to<std::string>,
+			std::allocator<std::pair<std::string, void*>>,
+			true
+		>;
+
+		using pinvoke_api_map_ptr = pinvoke_api_map*;
+		using pinvoke_library_map = tsl::robin_map<
+			std::string,
+			pinvoke_api_map_ptr,
+			std::hash<std::string>,
+			std::equal_to<std::string>,
+			std::allocator<std::pair<std::string, pinvoke_api_map_ptr>>,
+			true
+		>;
+
+		static constexpr pinvoke_library_map::size_type LIBRARY_MAP_INITIAL_BUCKET_COUNT = 1;
+#endif
```

Comment:
Convention (if not always followed): `#endif`/`#elif`/etc. should have a comment to mention what symbols they're related to:

```cpp
#endif  /* defined (NET6) */
```